bloom-housing/bloom
# feat: filter flagged duplicate sets

# Pull Request Template

## Issue Overview

This PR addresses #4100


## Description

Allows for the ability to search on application flagged sets by the first/last names of any applicant in a set.


Pending Set
The information in the first column was duplicated in the next two columns. Now the duplicate information is removed and we are showing the names of all applicants in the first column since there really is no "primary applicant" in a set and it was really confusing to search for an applicant name and have a line appear that didn't show that applicant's name, but because the applicant was in the set. 
Before:
![Screenshot 2024-05-21 at 1 11 28 PM](https://github.com/bloom-housing/bloom/assets/65367387/4ca4e554-be6e-43a4-b5a8-fe367d66383b)
After:
![Screenshot 2024-05-21 at 1 11 33 PM](https://github.com/bloom-housing/bloom/assets/65367387/7406646e-e4fc-4933-b540-aedc970799eb)

Resolved Set
Similar changes, duplicate information is removed and all applicant names are being shown
Before:
![Screenshot 2024-05-21 at 1 12 00 PM](https://github.com/bloom-housing/bloom/assets/65367387/6a3a3f94-65a2-4348-ada8-89990d484d1c)
After:
![Screenshot 2024-05-21 at 1 12 14 PM](https://github.com/bloom-housing/bloom/assets/65367387/306d72e8-ff95-4572-a524-34a0e0c15dc0)


Additionally, because that first column now needs to take up lots of space, I had to make adjustments to the sizes of the columns. I saw that the column sizes were wonky on all partners tables, so I lightly adjusted the spacings based on what I learned in the flagged set table.


## How Can This Be Tested/Reviewed?

Go to a listing with pending and resolved sets of application duplicates and try to search for names.

You can also look at the listing and application tables if you want to see some new column sizes. It's fairly subtle there.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
